create_builder.go

```diff
@@ -22,14 +22,24 @@ import (
 
 // CreateBuilderOptions are options passed to CreateBuilder
 type CreateBuilderOptions struct {
+	// Name of the builder
 	BuilderName string
-	Config      pubbldr.Config
-	Publish     bool
-	Registry    string
-	PullPolicy  config.PullPolicy
+
+	// Configuration that defines the functionality a builder provides.
+	Config pubbldr.Config
+
+	// Push resulting builder image up to Registry/BuilderName
```

Comment:
Instead of building an entire image locally and then pushing it, we create layers locally and update the image configuration in the registry.